File: codeforces/160A/cpp/160a.cpp
#include <iostream>
using namespace std;

const int c_max = 100;

int calc(int *count, int sum)
{
    int cur = 0, num = 0;
    for (int i = c_max; i >= 0; i--) {
        for (int j = 0; j < count[i]; j++) {
            cur += i;
            num++;
            if (cur > sum - cur)
                return num;
        }
    }

    return num;
}

int main()
{
    int n;
    cin >> n;

    int count[c_max + 1] = {0}, coin, sum = 0;
    for (int i = 0; i < n; i++) {
        cin >> coin;
        count[coin]++;
        sum += coin;
    }

    cout << calc(count, sum) << endl;

    return 0;
}

File: codeforces/160A/go/160a.go
package main

import "fmt"

func main() {
    var n int
    fmt.Scanf("%d\n", &n)

    var coin, sum int
    sum = 0
    var count = [101]int{0:101}
    for i := 0; i < n; i++ {
        fmt.Scanf("%d", &coin)
        sum += coin
        count[coin]++
    }

    cur := 0
    num := 0
    find := false
    for i := 100; i >= 0 && !find; i-- {
        for j := 0; j < count[i]; j++ {
            cur += i
            num++
            if (cur > sum - cur) {
                find = true
                break
            }
        }
    }

    fmt.Println(num)
}
